### PR TITLE
hack/stabilization-changes: Add support for day durations

### PR DESCRIPTION
**Pull request by wking**
From [RFC 3339][1]:

```
Durations:

dur-second        = 1*DIGIT "S"
dur-minute        = 1*DIGIT "M" [dur-second]
dur-hour          = 1*DIGIT "H" [dur-minute]
dur-time          = "T" (dur-hour / dur-minute / dur-second)
dur-day           = 1*DIGIT "D"
dur-week          = 1*DIGIT "W"
dur-month         = 1*DIGIT "M" [dur-day]
dur-year          = 1*DIGIT "Y" [dur-month]
dur-date          = (dur-day / dur-month / dur-year) [dur-time]

duration          = "P" (dur-date / dur-time / dur-week)
```

So fully supported:

* `dur-hour`
* `dur-day` (this commit)
* `dur-week`

Not supported:

* `dur-second`
* `dur-minute`
* `dur-month`
* `dur-year`

Partially supported:

* `dur-time`
* `dur-date`
* `duration`

This allows us to revert 75c9c782a6 (#1103), to regain the more-intuitive days.

[1]: https://datatracker.ietf.org/doc/html/rfc3339#page-13

### CODE REVIEW

File: hack/stabilization-changes.py [hunk 1]
```diff
@@ -33,10 +33,11 @@ def parse_iso8601_delay(delay):
     # https://tools.ietf.org/html/rfc3339#page-13
     match = _ISO_8601_DELAY_REGEXP.match(delay)
     if not match:
-        raise ValueError('invalid or unsupported ISO 8601 duration {!r}.  Tooling currently only supports P<number>WT<number>H for week and/or hour offsets')
+        raise ValueError('invalid or unsupported ISO 8601 duration {!r}.  Tooling currently only supports P<number>W for weeks, or P<number>DT<number>H for day/hour offsets'.format(delay))
```
Comment (LalatenduMohanty) [inline hack/stabilization-changes.py:36]: I wonder if we have any python library to do the same thing rather than doing it ourself.

Comment (wking) [reply hack/stabilization-changes.py:36]: I could not find one back when I initially implemented this script, but I'm not against slotting one in if someone else knows of one.

Comment (vrutkovs) [reply hack/stabilization-changes.py:36]: Two more projects - [arrow](https://github.com/arrow-py/arrow) and [pyiso8610](https://github.com/micktwomey/pyiso8601) - are parsing timestamps, but not durations. Perhaps we should pick one and contribute parsing code there?

Comment (vrutkovs) [reply hack/stabilization-changes.py:36]: Agree, its probably not worth it at this point

File: hack/stabilization-changes.py [hunk 2]
```diff
@@ -33,10 +33,11 @@ def parse_iso8601_delay(delay):
...
     weeks = int(match.group('weeks') or 0)
+    days = int(match.group('days') or 0)
     hours = int(match.group('hours') or 0)
-    return datetime.timedelta(days=7*weeks, hours=hours)
+    return datetime.timedelta(days=7*weeks + days, hours=hours)
```
Comment (vrutkovs) [inline hack/stabilization-changes.py:40]: `timedelta` has [weeks](https://docs.python.org/3.8/library/datetime.html#datetime.timedelta)

**Pull request merged**